plugins/release-manager-as-a-service/src/ReleaseManagerAsAService.tsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Alert } from '@material-ui/lab';
+import { CircularProgress, makeStyles } from '@material-ui/core';
+import { useAsync } from 'react-use';
+import React, { useState } from 'react';
+import { useApi, ContentHeader, ErrorBoundary } from '@backstage/core';
+
+import { CreateRc } from './cards/createRc/CreateRc';
+import { getGitHubBatchInfo } from './sideEffects/getGitHubBatchInfo';
+import { Info } from './cards/info/Info';
+import { Patch } from './cards/patchRc/Patch';
+import {
+  ComponentConfigCreateRc,
+  ComponentConfigPatch,
+  ComponentConfigPromoteRc,
+  GhGetBranchResponse,
+  GhGetReleaseResponse,
+  GhGetRepositoryResponse,
+  Project,
+  SetRefetch,
+} from './types/types';
+import { PromoteRc } from './cards/promoteRc/PromoteRc';
+import { releaseManagerAsAServiceApiRef } from './api/serviceApiRef';
+import {
+  ApiClientContext,
+  useApiClientContext,
+} from './components/ProjectContext';
+import { RMaaSApiClient } from './api/RMaaSApiClient';
+
+interface ReleaseManagerAsAServiceProps {
+  project: Project;
+  components?: {
+    default?: {
+      createRc?: ComponentConfigCreateRc;
+      promoteRc?: ComponentConfigPromoteRc;
+      patch?: ComponentConfigPatch;
+    };
+    custom?: ({
+      project,
+      setRefetch,
+      latestRelease,
+      releaseBranch,
+      repository,
+    }: {
+      project: Project;
+      setRefetch: SetRefetch;
+      latestRelease: GhGetReleaseResponse | null;
+      releaseBranch: GhGetBranchResponse | null;
+      repository: GhGetRepositoryResponse;
+    }) => JSX.Element[];
+  };
+}
+
+const useStyles = makeStyles(() => ({
+  root: {
+    maxWidth: '999px',
+  },
+}));
+
+export function ReleaseManagerAsAService({
+  project,
+  components,
+}: ReleaseManagerAsAServiceProps) {
+  const pluginApiClient = useApi(releaseManagerAsAServiceApiRef);
+  const RMaaSApi = new RMaaSApiClient({
+    pluginApiClient,
+    repoPath: `${project.github.org}/${project.github.repo}`,
+  });
+  const classes = useStyles();
+
+  return (
+    <ApiClientContext.Provider value={RMaaSApi}>
+      <div className={classes.root}>
+        <ContentHeader title="Release Manager as a Service (RMaaS™️)" />
```

Comment:
I would suggest this is not likely trademarked as a generic acronym, so recommend removing the `TM`